cat-launcher/src-tauri/src/last_played_world/commands.rs

```diff
@@ -0,0 +1,30 @@
+use strum::IntoStaticStr;
+use tauri::{command, AppHandle};
+
+use crate::{
+  last_played_world::last_played_world::{
+    get_last_played_world as get_last_played_world_impl,
+    GetLastPlayedWorldError,
+  },
+  variants::GameVariant,
+};
+use cat_macros::CommandErrorSerialize;
+
+#[derive(
+  thiserror::Error, Debug, IntoStaticStr, CommandErrorSerialize,
+)]
+pub enum GetLastPlayedWorldCommandError {
+  #[error("failed to get last played world: {0}")]
+  GetLastPlayedWorld(#[from] GetLastPlayedWorldError),
+}
+
+#[command]
+pub async fn get_last_played_world(
+  app_handle: AppHandle,
+  variant: GameVariant,
+) -> Result<Option<String>, GetLastPlayedWorldCommandError> {
+  let last_played_world =
+    get_last_played_world_impl(&app_handle, &variant).await?;
```

Comment:
app_handle is a Tauri framework related concept. It should not be exposed to business logic functions. Extract what is needed from the app_handle and only pass those as parameters to the get_last_played_world_impl function.